pkg/controller/networkpolicy/validate.go

```diff
@@ -629,6 +629,18 @@ func numFieldsSetInPeer(peer crdv1alpha1.NetworkPolicyPeer) int {
 	return num
 }
 
+// numFieldsSetInAppliedTo returns the number of fields in use of the object.
+func numFieldsSetInAppliedTo(obj interface{}) int {
+	num := 0
+	v := reflect.ValueOf(obj)
+	for i := 0; i < v.NumField(); i++ {
+		if !v.Field(i).IsZero() {
+			num++
+		}
+	}
+	return num
+}
+
```

Comment:
I meant this could replace the above `numFieldsSetInPeer` so it only needs one function to process all structs.

Reply:
@qiyueyao could we address this?